FloripaBus/Repositories/IRouteRepository.cs

```diff
@@ -0,0 +1,168 @@
+﻿using System;
+using System.Threading.Tasks;
+using System.Collections.Generic;
+using System.Net.Http;
+using System.Net.Http.Headers;
+using System.Text;
+using Newtonsoft.Json;
+using System.Net;
+
+namespace FloripaBus
+{
+	public interface IRouteRepository
+	{
+		Task<IList<Route>> FindRoutesByStopNameAsync(string stopName);
+		Task<IList<RouteStop>> FindStopsByRouteIdAsync(int routeId);
+		Task<IList<RouteDeparture>> FindDeparturesByRouteIdAsync(int routeId);
+	}
+
+	public class RouteRepository : IRouteRepository
+	{		
+		private const string USER = "WKD4N7YMA1uiM8V";
+		private const string PASSWORD = "DtdTtzMLQlA0hk2C1Yi5pLyVIlAQ68";
+		private const string FIND_ROUTES_URL = "https://api.appglu.com/v1/queries/findRoutesByStopName/run";
+		private const string FIND_STOPS_URL = "https://api.appglu.com/v1/queries/findStopsByRouteId/run";
+		private const string FIND_DEPARTURES_URL = "https://api.appglu.com/v1/queries/findDeparturesByRouteId/run";
+
+		private readonly HttpClient _httpClient;
+
+		public RouteRepository(HttpClient httpClient)
+		{
+			_httpClient = httpClient;
+		}
+
+		public async Task<IList<Route>> FindRoutesByStopNameAsync (string stopName)
+		{
+			IList<Route> routes = new List<Route> ();
+
+			string responseContent;
+			using (var request = new HttpRequestMessage (HttpMethod.Post, FIND_ROUTES_URL)) {
+				
+				AddDefaultHeaders (request);
+
+				var body = new StringBuilder ()

```

Comment:
The string which is being built hás a few of '{' carácter that causes an issue to the string.Format method.